borg/cache.py

```diff
@@ -140,20 +142,39 @@ def create(self):
         with open(os.path.join(self.path, 'files'), 'wb') as fd:
             pass  # empty file
 
-    def _do_open(self):
-        self.config = configparser.ConfigParser(interpolation=None)
-        config_path = os.path.join(self.path, 'config')
-        self.config.read(config_path)
+    def _check_upgrade(self, config_path):
         try:
             cache_version = self.config.getint('cache', 'version')
-            wanted_version = 1
-            if cache_version != wanted_version:
-                self.close()
-                raise Exception('%s has unexpected cache version %d (wanted: %d).' % (
-                    config_path, cache_version, wanted_version))
+            if cache_version != CACHE_VERSION:
+                if cache_version == 1:
+                    # borg < 1.0.8rc1 (cache version 1) had different canonicalization,
+                    # upgrade it to current canonicalization (cache version 2):
+                    cache_loc = self.config.get('cache', 'previous_location', fallback=None)
+                    if cache_loc:
+                        repo_loc = self.repository._location.canonical_path()
+                        rl = Location(repo_loc)
+                        cl = Location(cache_loc)
+                        if cl.proto == rl.proto and cl.user == rl.user and cl.host == rl.host and cl.port == rl.port \
+                           and \
+                           cl.path and rl.path and \
+                           cl.path.startswith('/~/') and rl.path.startswith('/./') and cl.path[3:] == rl.path[3:]:
+                            # everything is same except the expected change in relative path canonicalization,
+                            # update previous_location to avoid warning / user query about changed location:
+                            self.config.set('cache', 'previous_location', repo_loc)
```

Comment:
I don't think that this is a security feature; the server can remap paths however it likes. So this is a check to avoid ... mixups perhaps? So I think it wouldn't be a problem.